# Determination - Deterministic rendering environment for white-axe's music
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# Mephisto for some reason doesn't initialize static tables in Faust programs.
# That means programs that use the `rdtable`, `rwtable` or `waveform` Faust
# primitives outright won't work. This patch remedies that.

--- a/src/mephisto.c
+++ b/src/mephisto.c
@@ -2254,7 +2254,7 @@ _dsp_init(plughandle_t *handle, dsp_t *dsp, const char *code,
 		goto fail;
 	}
 
-	instanceInitCDSPInstance(base_voice->instance, handle->srate);
+	initCDSPInstance(base_voice->instance, handle->srate);
 
 	dsp->nins = getNumInputsCDSPInstance(base_voice->instance);
 	dsp->nouts = getNumInputsCDSPInstance(base_voice->instance);
